WooCommerce/Classes/POS/Presentation/Card Present Payments/PointOfSaleCardPresentPaymentEventPresentationStyle.swift

```diff
@@ -5,11 +5,19 @@ enum PointOfSaleCardPresentPaymentEventPresentationStyle {
     case alert(PointOfSaleCardPresentPaymentAlertType)
 }
 
-/// View Models are created here, but can be "annotated" where they are used if the `CardPresentPaymentEventDetails` is
-/// not enough to fully populate the view model. See `TotalsViewModel.observeCardPresentPaymentEvents` for an example.
-extension CardPresentPaymentEventDetails {
-    var pointOfSalePresentationStyle: PointOfSaleCardPresentPaymentEventPresentationStyle? {
-        switch self {
+/// A factory for payment alert/message view models, so we can provide information beyond what's available from  `CardPresentPaymentEventDetails`
+/// To avoid keeping the additional information up to date as it changes, we pass it to every call to `presentationStyle`
+/// See `TotalsViewModel.observeCardPresentPaymentEvents` for an example.
+struct PointOfSaleCardPresentPaymentEventPresentationStyleDeterminer {
+    struct Dependencies {
+        let tryPaymentAgainBackToCheckoutAction: () -> Void
+        let nonRetryableErrorExitAction: () -> Void
+        let formattedOrderTotalPrice: String?
+    }
+
+    static func presentationStyle(for eventDetails: CardPresentPaymentEventDetails,
+                                  dependencies: Dependencies) -> PointOfSaleCardPresentPaymentEventPresentationStyle? {
+        switch eventDetails {
```

Comment:
Perhaps this should actually be an init on `PointOfSaleCardPresentPaymentEventPresentationStyle` – I had it in a new object because it kept some state, originally, but I decided that was a bit risky and went for static to avoid it.

Reply:
Done in 4fe3e0a